src/app/calendar/ParcelCalendar.tsx

```diff
@@ -13,20 +28,44 @@ const Centerer = styled.div`
 
 const CalendarWrapper = styled.div`
     width: 100vmin;
-    margin: 2em;
+    margin: 0 2em 2em;
 `;
 
-interface ParcelCalendarProps {
-    events: CalendarEvent[];
-}
-
 const ParcelCalendar: React.FC<ParcelCalendarProps> = (props) => {
+    const chosenTheme = useTheme();
+
+    const getColorText = (color: keyof rainbowPalette): ColorTextProps => {
+        return {
+            color: `${chosenTheme.rainbow[color].background}`,
+            text: `${chosenTheme.rainbow[color].foreground}`,
+        };
+    };
+
+    const colorMap: LocationColorMap = {
+        "Brixton Hill - Methodist Church": getColorText("lightRed"),
+        "Clapham - St Stephens Church": getColorText("lightOrange"),
+        "N&B - Emmanuel Church": getColorText("lightYellow"),
+        "Streatham - Immanuel & St Andrew": getColorText("lightGreen"),
+        "Vauxhall Hope Church": getColorText("darkGreen"),
+        "Waterloo - Oasis": getColorText("lightBlue"),
+        "Waterloo - St George the Martyr": getColorText("darkBlue"),
+        "Waterloo - St Johns": getColorText("lightPurple"),
+        Delivery: getColorText("darkPurple"),
+        default: getColorText("lightBrown"),
+    };
     return (
         <>
             <Title>Collection Time for Parcels</Title>
             <Centerer>
                 <CalendarWrapper>
-                    <Calendar initialEvents={props.events} editable={false} />
+                    <Calendar
+                        initialEvents={parcelsToCollectionEvents(
```

Comment:
It is because we have to read the color for the color map from the theming, which has to make use of `useTheme()`
`useTheme()` is a hook thing which has to be run on client side, which is used to generate `colorMap`, a necessary prop for the function `parcelsToCollectionEvents` . (Originally the `colorMap` is just hard-coded so it exists on server side)
Hence now, `parcelsToCollectionEvents` has to be moved to the client side and only be run when `colorMap` is ready on client side.
The change is therefore necessary

Reply:
Initially, `props.event` is from the calendar page, which runs the function `parcelsToCollectionEvents` from the calendar page. The calendar page is purely server-side rendered, and now because of the existence of the colorMap, `parcelsToCollectionEvents` can no longer be run on the server-side but the client-side. Hence we can't read props.event from the server-side page directly, but the function to fetch the event has to be run in `ParcelCalendar.tsx` which is purely client-side